### PR TITLE
Add statistics endpoint for CategoryViewSet

### LINKED ISSUE #10504: No /statistics REST API for categories
### Describe the problem

The UI is displaying a nice view to show how much a language is translated for a given category:
/projects/:project/:category/#languages

However the REST API does not provide such route.

### Describe the solution you'd like

I guess we will need a new route GET /categories/:category_id/statistics/

### Describe alternatives you've considered

I am currently web scrapping from the UI.

### Screenshots

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by metalglove**
## Proposed changes
Added the statistics endpoint for the CategoryViewSet component. As discussed in #10504 

## Checklist

- [x] Lint and unit tests pass locally with my changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added documentation to describe my feature.
- [x] I have squashed my commits into logic units.
- [x] I have described the changes in the commit messages.

## Other information
Fixes #10504

### CODE REVIEW

**Comment by metalglove** [comment]
For the documentation, I see that there are also notes regarding in which version a piece of functionality is added. 
I have left it out for now. As I am not sure for which version this functionality will be included, if at all.

Example of what I mean:
`.. versionadded:: 5.0`

**Comment by metalglove** [comment]
I have not used Django enough to completely understand how to get my test to pass.

To my understanding, the Category model has `id` as primary key. (that is also how it is used in the api call `/api/categories/{id:int}/statistics/`).
So, I used that as the link to the model for the category statistics test. However, this does not work.
I could use some guidance to get this test to pass. What have I missed?

My error boils down to
```
django.core.exceptions.ImproperlyConfigured: Could not resolve URL for hyperlinked relationship 
using view name "api:category-statistics". You may have failed to include the related model 
in your API, or incorrectly configured the `lookup_field` attribute on this field.
```

**Comment by metalglove** [comment]
I have managed to make the test pass. 🥳

All that is left is to make sure that the documentation has that correct version mention `.. versionadded:: 5.5`?

File: docs/api.rst [hunk 1]
```diff
@@ -2597,6 +2598,17 @@ Categories
     :param id: Category ID
     :type id: int
 
+.. http:get:: /api/categories/(int:id)/statistics/
+
+    Returns statistics for a category.
```
Comment (nijel) [inline docs/api.rst:2605]:
```suggestion
    .. versionadded:: 5.5
  
    Returns statistics for a category.
```

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**